program/src/instruction.rs

```diff
@@ -1,184 +1,373 @@
-//! Program instructions
+use borsh::{BorshDeserialize, BorshSerialize};
+use shank::{ShankContext, ShankInstruction};
 
-use solana_program::clock::UnixTimestamp;
-
-/// Instructions supported by the staking program
-#[derive(Clone, Debug, PartialEq, Eq)]
-pub enum StakeInstruction {
-    /// Creates Stake config account which controls staking parameters
-    ///
-    /// 0. `[w]` Stake config account
-    /// 1. `[]` Slash authority, may be a PDA for a governance program
-    /// 2. `[]` Config authority, may be a PDA for a governance program
-    /// 3. `[]` Stake Token Mint
-    /// 4. `[]` Stake Token Vault, to hold all staked tokens.
-    ///   Must be empty and owned by PDA with seeds `['token-owner', stake_config]`
+/// Enum defining all instructions in the Stake program.
+#[rustfmt::skip]
+#[derive(BorshDeserialize, BorshSerialize, Clone, Debug, ShankContext, ShankInstruction)]
+pub enum Instruction {
```

Comment:
Can we keep this named `StakeInstruction`?

Reply:
Done.